src/casts/coercion-passing.rkt

```diff
@@ -0,0 +1,310 @@
+       (apply-coercion-opt
+        (Blame (trans-exp v ID-EXPR))
+        cont)]
+
+      ;; box (references):
+      [(Unguarded-Box v)
+       (apply-coercion-opt
+        (Unguarded-Box (trans-exp v ID-EXPR))
+        cont)]
+      [(Unguarded-Box-Ref box)
+       (apply-coercion-opt
+        (Unguarded-Box-Ref (trans-exp box ID-EXPR))
+        cont)]
+      [(Unguarded-Box-Set! box v)
+       (apply-coercion-opt
+        (Unguarded-Box-Set! (trans-exp box ID-EXPR) (trans-exp v ID-EXPR))
+        cont)]
+
+      ;; monobox:
+      [(Mbox v ty)
+       (apply-coercion-opt
+        (Mbox (trans-exp v ID-EXPR) ty)
+        cont)]
+      [(Mbox-val-ref e)
+       (apply-coercion-opt
+        (Mbox-val-ref (trans-exp e ID-EXPR))
+        cont)]
+      [(Mbox-val-set! e1 e2)
+       (apply-coercion-opt
+        (Mbox-val-set! (trans-exp e1 ID-EXPR) (trans-exp e2 ID-EXPR))
+        cont)]
+      [(Mbox-rtti-ref e)
+       (apply-coercion-opt
+        (Mbox-rtti-ref (trans-exp e ID-EXPR))
+        cont)]
+
+      ;; vectors:
+      [(Unguarded-Vect e1 e2)
+       (apply-coercion-opt
+        (Unguarded-Vect (trans-exp e1 ID-EXPR) (trans-exp e2 ID-EXPR))
+        cont)]
+      [(Unguarded-Vect-Ref e1 e2)
+       (apply-coercion-opt
+        (Unguarded-Vect-Ref (trans-exp e1 ID-EXPR) (trans-exp e2 ID-EXPR))
+        cont)]
+      [(Unguarded-Vect-Set! v1 v2 v3)
+       (apply-coercion-opt
+        (Unguarded-Vect-Set! (trans-exp v1 ID-EXPR) (trans-exp v2 ID-EXPR) (trans-exp v3 ID-EXPR))
+        cont)]
+      [(Unguarded-Vect-length v)
+       (apply-coercion-opt
+        (Unguarded-Vect-length (trans-exp v ID-EXPR))
+        cont)]
+
+      ;; proxied reference coercions:
+      [(Ref-Coercion-Read e)
+       (apply-coercion-opt
+        (Ref-Coercion-Read (trans-exp e ID-EXPR))
+        cont)]
+      [(Ref-Coercion-Write e)
+       (apply-coercion-opt
+        (Ref-Coercion-Write (trans-exp e ID-EXPR))
+        cont)]
+
+      ;; tuples:
+      [(Create-tuple e*)
+       (apply-coercion-opt
+        (Create-tuple
+         (for/list ([a e*])
+           (trans-exp a ID-EXPR)))
+        cont)]
+      [(Tuple-proj e1 e2)
+       (apply-coercion-opt
+        (Tuple-proj (trans-exp e1 ID-EXPR) (trans-exp e2 ID-EXPR))
+        cont)]
+
+      ;; proxies:
+      [(Guarded-Proxy-Huh v)
+       (apply-coercion-opt
+        (Guarded-Proxy-Huh (trans-exp v ID-EXPR))
+        cont)]
+      [(Guarded-Proxy-Ref v)
+       (apply-coercion-opt
+        (Guarded-Proxy-Ref (trans-exp v ID-EXPR))
+        cont)]
+      [(Guarded-Proxy-Coercion v)
+       (apply-coercion-opt
+        (Guarded-Proxy-Coercion (trans-exp v ID-EXPR))
+        cont)]
+
+      [(Type v)
+       (Type v)]
+
+      [(Quote-Coercion c)
+       (Quote-Coercion c)]
+      [(Sequence-Coercion e1 e2)
+       (Sequence-Coercion e1 e2)]
+      [(Project-Coercion e1 e2)
+       (Project-Coercion e1 e2)]
+
+      ;; forms whose translations are unimplemented:
+      [_ (pretty-print exp)
```

Comment:
Since I got at-mentioned I'll just sort of randomly add that the changes I made to get this improved error-reporting were in the C-backend for Racket... I haven't yet double checked if the CS backend has it. Anyway, have a good day! =)

Reply:
OK. I commented out this part. 9e3fc6a